test/e2e/components/vl-select.js

```diff
@@ -135,58 +179,68 @@ class VlSelect extends VlElement {
     return parent.findElement(By.css('button.vl-pill__close'));
   }
 
-  async _getOptions() {
-    if (await this.isDressed()) {
-      const selectList = await this._getSelectList();
-      return selectList.findElements(By.css('.vl-select__item'));
-    } else {
-      return this.findElements(By.css('option'));
+  async _clickOption(option) {
+    if ((await this.isDressed())) {
+      await this.open();
     }
+    return option.click();
   }
+}
 
-  async _mapDressedOptions(options) {
-    return Promise.all(options.map(async (selectItem) => {
-      const text = await selectItem.getText();
-      const webElement = await selectItem;
-      return {text: text, webElement: webElement};
-    }));
+class OptionGroup {
+  constructor(groupItem, dressed, options) {
+	this.groupItem = groupItem;
+	this._options = options || [];
+	this.dressed = dressed;
+  }
+	
+  addOption(item) {
+    this._options.push(item)
+  }
+
+  get label() {
+	if (this.dressed) {
+	  return this.groupItem.findElement(By.css('.vl-select__heading')).then((heading) => {
+        return heading.getAttribute('textContent').then((textContent) => {
+	      return textContent.trim();
+        });	
+	  });
+	} else {
+	  return this.groupItem.getAttribute('label');	
+	}
+  }
```

Comment:
hier is een interessante discussie.
ge kunt geen async getters schrijven, dus ik had de keuze tussen ne getter met deze code, of code met async await en een methode ipv ne getter.

nog meningen hier?

Reply:
Eigenlijk is het dus ook async, maar we geven gewoon ne promise terug en markeren het niet als async.
Op zich dan iets liever ne "async getLabel()", doen we ook bij de rest eigenlijk op die manier.

Reply:
Ik zal het aanpassen dan.